mne/preprocessing/artifact_detection.py

```diff
@@ -0,0 +1,316 @@
+    chpi_locs_dev = sorted([d for d in info['hpi_results'][-1]
+                            ['dig_points']], key=lambda x: x['ident'])
+    chpi_locs_dev = np.array([d['r'] for d in chpi_locs_dev])
+
+    # Get head pos displacement
+    chpi_mov_head = np.array([_apply_quat(quat, chpi_locs_dev, move=True)
+                              for quat in quats])
+
+    # Get weights for each head pos
+    weights = np.append(time[1:] - time[:-1], 0)
+
+    # Bad segments have a weight of 0
+    for annoth in raw.annotations:
+        if 'bad_' in annoth['description']:
+            onset = annoth['onset']
+            end = onset + annoth['duration']
+            out_beg = np.argmin(np.abs(time - onset))
+            out_end = np.argmin(np.abs(time - end))
+            weights[out_beg:out_end] = 0
+    weights /= sum(weights)
+
+    # Get head pos closest to weighted average pos
+    avg_head_pos = np.average(chpi_mov_head, axis=0, weights=weights)
+    hpi_disp = chpi_mov_head - np.tile(avg_head_pos, (len(time), 1, 1))
+    hpi_dist = np.sqrt((hpi_disp.reshape(-1, 9) ** 2).sum(axis=1))
+    head_pos_avg = chpi_mov_head[hpi_dist.argmin(), :, :]
+
+    # Compute new dev->head transformation from median
+    dev_head_t = _quaternion_align(info['dev_head_t']['from'],
+                                   info['dev_head_t']['to'],
+                                   chpi_locs_dev, head_pos_avg)
+
+    return dev_head_t
+
+
+
+def detect_muscle(raw, thr=1.5, t_min=1, notch=[60, 120, 180]):
+    """Find and annotate mucsle artifacts based on high frequency activity.
+
+    Data is band pass filtered at high frequencies, smoothed with the envelope,
+    z-scored, channel averaged and low-pass filtered to remove transient peaks.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The data.
+    thr : integer
+        threshold where a segment is marked as artifactual. The thr represent
+        the channel averaged z-score values of the data band pass filtered
+        betweehn 110 and 140 Hz.
+    t_min = integer
+        Min. time between annotated muscle artifacts, below will be annotated
+        as muscle artifact.
+    notch = List
+        The frequencies at which to apply a notch filter
+
+    Returns
+    -------
+    annot : mne.Annotations
+        periods with muscle artifacts
+    art_scores_filt : array
+        the z-score values of the filtered data
+    """
+    raw.pick_types(meg=True, ref_meg=False)
+    if notch is not None:
+        raw.notch_filter(notch, fir_design='firwin')
+    raw.filter(110, 140, fir_design='firwin')
+    raw.apply_hilbert(envelope=True)
+    sfreq = raw.info['sfreq']
+    art_scores = zscore(raw._data, axis=1)
+    # band pass filter the data
+    art_scores_filt = filter_data(art_scores.mean(axis=0), sfreq, None, 4)
+    art_mask = art_scores_filt > thr
+    # remove artifact free periods shorter than t_min
+    idx_min = t_min * sfreq
+    comps, num_comps = label(art_mask == 0)
```

Comment:
why `art_mask == 0` and not `... == 1`? - doesn't `label` group "clusters" of `True`?

Reply:
This section if for annotating good segments too short

Reply:
ah, right - it says "artifact free"!